File: app/src/main/java/org/michaelbel/moviemade/data/di/component/FragmentComponent.java
package org.michaelbel.moviemade.data.di.component;

import org.michaelbel.moviemade.data.di.module.FragmentModule;
import org.michaelbel.moviemade.data.di.scope.FragmentScoped;
import org.michaelbel.moviemade.ui.modules.account.AccountFragment;
import org.michaelbel.moviemade.ui.modules.favorites.FavoritesFragment;
import org.michaelbel.moviemade.ui.modules.keywords.fragment.KeywordFragment;
import org.michaelbel.moviemade.ui.modules.keywords.fragment.KeywordsFragment;
import org.michaelbel.moviemade.ui.modules.main.fragments.NowPlayingFragment;
import org.michaelbel.moviemade.ui.modules.main.fragments.TopRatedFragment;
import org.michaelbel.moviemade.ui.modules.main.fragments.UpcomingFragment;
import org.michaelbel.moviemade.ui.modules.movie.BackdropDialog;
import org.michaelbel.moviemade.ui.modules.movie.MovieFragment;
import org.michaelbel.moviemade.ui.modules.recommendations.RcmdMoviesFragment;
import org.michaelbel.moviemade.ui.modules.reviews.fragment.ReviewFragment;
import org.michaelbel.moviemade.ui.modules.reviews.fragment.ReviewsFragment;
import org.michaelbel.moviemade.ui.modules.search.SearchMoviesFragment;
import org.michaelbel.moviemade.ui.modules.settings.SettingsFragment;
import org.michaelbel.moviemade.ui.modules.similar.SimilarMoviesFragment;
import org.michaelbel.moviemade.ui.modules.trailers.TrailersFragment;
import org.michaelbel.moviemade.ui.modules.watchlist.WatchlistFragment;

import dagger.Subcomponent;

@FragmentScoped
@Subcomponent(modules = FragmentModule.class)
public interface FragmentComponent {
    void inject(MovieFragment target);
    void inject(ReviewFragment target);
    void inject(AccountFragment target);
    void inject(FavoritesFragment target);
    void inject(SettingsFragment target);
    void inject(WatchlistFragment target);
    void inject(SimilarMoviesFragment target);
    void inject(RcmdMoviesFragment target);
    void inject(TrailersFragment target);
    void inject(NowPlayingFragment target);
    void inject(TopRatedFragment target);
    void inject(UpcomingFragment target);
    void inject(SearchMoviesFragment target);
    void inject(ReviewsFragment target);
    void inject(KeywordFragment target);
    void inject(KeywordsFragment target);
    void inject(BackdropDialog target);
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/di/component/FragmentComponent.kt
package org.michaelbel.moviemade.presentation.di.component

import dagger.Subcomponent
import org.michaelbel.moviemade.presentation.common.BottomSheetDialog
import org.michaelbel.moviemade.presentation.di.module.FragmentModule
import org.michaelbel.moviemade.presentation.di.scope.FragmentScope
import org.michaelbel.moviemade.presentation.features.keywords.KeywordsFragment
import org.michaelbel.moviemade.presentation.features.login.LoginFragment
import org.michaelbel.moviemade.presentation.features.main.MoviesFragment
import org.michaelbel.moviemade.presentation.features.main.MoviesFragment2
import org.michaelbel.moviemade.presentation.features.movie.MovieFragment
import org.michaelbel.moviemade.presentation.features.review.ReviewFragment
import org.michaelbel.moviemade.presentation.features.reviews.ReviewsFragment
import org.michaelbel.moviemade.presentation.features.search.SearchMoviesFragment
import org.michaelbel.moviemade.presentation.features.trailers.TrailersFragment
import org.michaelbel.moviemade.presentation.features.user.UserFragment

@FragmentScope
@Subcomponent(modules = [FragmentModule::class])
interface FragmentComponent {
    fun inject(target: MovieFragment)
    fun inject(target: ReviewFragment)
    fun inject(target: LoginFragment)
    fun inject(target: TrailersFragment)
    fun inject(target: MoviesFragment)
    fun inject(target: MoviesFragment2)
    fun inject(target: SearchMoviesFragment)
    fun inject(target: ReviewsFragment)
    fun inject(target: KeywordsFragment)
    fun inject(target: BottomSheetDialog)
    fun inject(target: UserFragment)
}